fsspec/implementations/arrow.py

```diff
@@ -61,6 +61,20 @@ def protocol(self):
     def fsid(self):
         return "hdfs_" + tokenize(self.fs.host, self.fs.port)
 
+    @classmethod
+    def from_fs(cls, fs, **kwargs):
+        override = {"local": "file"}
+        type_name = override.get(fs.type_name, fs.type_name)
+
+        try:
+            fs.from_uri(f"{type_name}:///")
```

Comment:
This is the foolproof way to know whether the first character can/should be "/"?

Reply:
It definitely works for (3/4) Local, S3, and GCS. I'm not sure about Hadoop, I'm never used it and am not sure how to test locally. 

https://arrow.apache.org/docs/python/filesystems.html#filesystem-interface

Reply:
hadoop has a derived class anyway, HadoopFileSystem. We should make sure it has "/", and it doesn't need to go through the check in this PR.